### DISCUSSION TITLE
reflect table doesn't get the partition column as an index in SQLAlchemy 2.0

### DISCUSSION THREAD

**Discussion by mdeshmu** [sqlalchemy/sqlalchemy#9997] [Usage Questions] [answered]
I have a table with the following table definition:

`CREATE TABLE many_rows (
    a INT
) PARTITIONED BY (
    b STRING
) ROW FORMAT DELIMITED FIELDS TERMINATED BY '\t' STORED AS TEXTFILE;
`

We have the following test case which passes with SQLAlchemy 1.3 and 1.4 but fails with SQLAlchemy 2.0. 

`
many_rows = Table('many_rows', MetaData(), autoload_with=engine)
self.assertEqual(repr(many_rows.indexes), repr({Index('partition', many_rows.c.b)}))
`

Here is the test case output in 2.0

`
self.assertEqual(repr(many_rows.indexes), repr({Index('partition', many_rows.c.b)}))    
E   AssertionError: 'set()' != "{Index('partition', Column('b', String(), table=<many_rows>))}"  
E   - set()
E   + {Index('partition', Column('b', String(), table=<many_rows>))}
`

My expectation is reflected table should get the partition column as an index but in SQLAlchemy 2.0, I get an empty set. 
Even inspector.get_indexes also gives an empty set. 

@zzzeek and team, Can someone please explain me this 2.0 behavior. 

Dialect: https://github.com/dropbox/PyHive/blob/master/pyhive/sqlalchemy_presto.py#L170
Test Case: https://github.com/dropbox/PyHive/blob/master/pyhive/tests/sqlalchemy_test_case.py#L82

**Comment by zzzeek** [comment]
@CaselIT have the queries for PG reflection possibly lost compatibility with this table?

**Reply by CaselIT** [reply]
that's very likely the case. I don't think we have any case that tests partitioned tables.

I'll look at this better later today

**Comment by CaselIT** [comment, marked as answer]
Hi,

I cannot reproduce using postgresql only. You should probably report the issue on the presto dialect
Test scropt:
```py
import sqlalchemy as sa
from pprint import pprint

e = sa.create_engine("postgresql+psycopg2://scott:tiger@127.0.0.1:5432/test")


def tbl():
    return """
    create table x(id int unique);

    create table with_partition (
        a INT references x (id),
        b int primary key unique,
        check (a>0),
        unique (a,b)
    ) partition by range (b);

    comment on table with_partition is 'comment';

    create index wpi on with_partition(a);

    create table no_partition (
        a INT references x (id),
        b int primary key,
        check (a>0),
        unique (a,b)
    );

    comment on table no_partition is 'comment';

    create index npi on no_partition(a);
    """


with e.connect() as c:
    c.exec_driver_sql(tbl())
    i = sa.inspect(c)
    pprint(i.get_table_names())
    pprint(i.get_multi_check_constraints())
    pprint(i.get_multi_columns())
    pprint(i.get_multi_indexes())
    pprint(i.get_multi_foreign_keys())
    pprint(i.get_multi_pk_constraint())
    pprint(i.get_multi_table_comment())
    pprint(i.get_multi_unique_constraints())
```

**Reply by mdeshmu** [reply]
It was indeed a dialect issue.
